addons/portal/controllers/portal.py

```diff
@@ -31,20 +36,39 @@ def pager(url, total, page=1, step=30, scope=5, url_args=None):
     page_count = int(math.ceil(float(total) / step))
 
     page = max(1, min(int(page if str(page).isdigit() else 1), page_count))
-    scope -= 1
-
-    pmin = max(page - int(math.floor(scope/2)), 1)
-    pmax = min(pmin + scope, page_count)
 
-    if pmax - pmin < scope:
-        pmin = pmax - scope if pmax - scope > 0 else 1
+    page_previous = max(1, page - 1)
+    page_next = min(page_count, page + 1)
 
     def get_url(page):
         _url = "%s/page/%s" % (url, page) if page > 1 else url
         if url_args:
             _url = "%s?%s" % (_url, urls.url_encode(url_args))
         return _url
 
+    def add_pages(page_list):
+        """ Helper to add a page or ellipsis to the pages list. """
+        for current_page in page_list:
```

Comment:
variable name current_page is misleading only p is fine if page is already reserved